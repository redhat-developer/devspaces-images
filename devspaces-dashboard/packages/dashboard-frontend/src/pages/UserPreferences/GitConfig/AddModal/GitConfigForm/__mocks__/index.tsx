/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/UserPreferences/GitConfig/AddModal/GitConfigForm';
import * as GitConfigStore from '@/store/GitConfig';

export class GitConfigForm extends React.PureComponent<Props> {
  render() {
    const { onChange } = this.props;

    return (
      <div>
        <input
          data-testid="submit-invalid-git-config"
          type="button"
          onClick={() =>
            onChange({ user: { name: 'User One' } } as GitConfigStore.GitConfig, false)
          }
        />
        <input
          data-testid="submit-valid-git-config"
          type="button"
          onClick={() =>
            onChange({ user: { email: 'user-1@chetest.com', name: 'User One' } }, true)
          }
        />
      </div>
    );
  }
}
