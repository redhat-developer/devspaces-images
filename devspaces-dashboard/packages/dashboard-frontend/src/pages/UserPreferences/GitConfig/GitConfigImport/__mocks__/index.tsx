/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/UserPreferences/GitConfig/GitConfigImport';

export class GitConfigImport extends React.PureComponent<Props> {
  render() {
    const { content, onChange } = this.props;

    return (
      <div>
        <input
          data-testid="submit-invalid-git-config"
          type="button"
          value={content}
          onClick={() => onChange('[user]\n\tname = User One\n', false)}
        />
        <input
          data-testid="submit-valid-git-config"
          type="button"
          onClick={() =>
            onChange('[user]\n\tname = User One\n\temail = user-1@chetest.com\n', true)
          }
        />
      </div>
    );
  }
}
