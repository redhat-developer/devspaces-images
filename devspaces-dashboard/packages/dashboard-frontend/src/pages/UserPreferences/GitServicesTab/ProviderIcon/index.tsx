/*
 * Copyright (c) 2018-2023 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { api } from '@eclipse-che/common';
import {
  CheckCircleIcon,
  ExclamationTriangleIcon,
  ResourcesEmptyIcon,
} from '@patternfly/react-icons';
import React from 'react';
import { connect, ConnectedProps } from 'react-redux';

import { CheTooltip } from '@/components/CheTooltip';
import { AppState } from '@/store';
import * as GitOauthConfig from '@/store/GitOauthConfig';
import {
  selectProvidersWithToken,
  selectSkipOauthProviders,
} from '@/store/GitOauthConfig/selectors';

type State = {
  hasOauthToken: boolean;
  isSkipOauth: boolean;
};

type Props = MappedProps & {
  gitProvider: api.GitOauthProvider;
};

export class ProviderIcon extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);
    const hasOauthToken = this.hasOauthToken(this.props.gitProvider);
    const isSkipOauth = this.isSkipOauth(this.props.gitProvider);
    this.state = {
      hasOauthToken,
      isSkipOauth,
    };
  }

  public async componentDidMount(): Promise<void> {
    const hasOauthToken = this.hasOauthToken(this.props.gitProvider);
    const isSkipOauth = this.isSkipOauth(this.props.gitProvider);
    this.setState({
      hasOauthToken,
      isSkipOauth,
    });
  }

  public async componentDidUpdate(): Promise<void> {
    const hasOauthToken = this.hasOauthToken(this.props.gitProvider);
    const isSkipOauth = this.isSkipOauth(this.props.gitProvider);
    this.setState({
      hasOauthToken,
      isSkipOauth,
    });
  }

  private isSkipOauth(providerName: api.GitOauthProvider): boolean {
    return this.props.skipOauthProviders.includes(providerName);
  }

  private hasOauthToken(providerName: api.GitOauthProvider): boolean {
    return this.props.providersWithToken.includes(providerName);
  }

  public render(): React.ReactElement {
    const { hasOauthToken, isSkipOauth } = this.state;
    if (hasOauthToken) {
      return (
        <CheTooltip content={<span>User has been authorized successfully.</span>}>
          <CheckCircleIcon color="var(--pf-global--success-color--100)" />
        </CheTooltip>
      );
    } else if (isSkipOauth) {
      return (
        <CheTooltip content={<span>Authorization has been rejected by user.</span>}>
          <ExclamationTriangleIcon color="var(--pf-global--warning-color--100)" />
        </CheTooltip>
      );
    }

    return (
      <CheTooltip content={<span>User has not been authorized yet.</span>}>
        <ResourcesEmptyIcon color="var(--pf-global--disabled-color--100)" />
      </CheTooltip>
    );
  }
}

const mapStateToProps = (state: AppState) => ({
  providersWithToken: selectProvidersWithToken(state),
  skipOauthProviders: selectSkipOauthProviders(state),
});

const connector = connect(mapStateToProps, GitOauthConfig.actionCreators);

type MappedProps = ConnectedProps<typeof connector>;
export default connector(ProviderIcon);
