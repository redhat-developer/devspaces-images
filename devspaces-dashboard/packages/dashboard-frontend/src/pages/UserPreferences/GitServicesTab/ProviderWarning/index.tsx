/*
 * Copyright (c) 2018-2023 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { WarningTriangleIcon } from '@patternfly/react-icons';
import React from 'react';

import { CheTooltip } from '@/components/CheTooltip';

type Props = {
  serverURI: string;
};
export default class ProviderWarning extends React.PureComponent<Props> {
  public render(): React.ReactElement {
    const content = (
      <>
        Provided API does not support the automatic token revocation. You can revoke it manually on
        &nbsp;
        <a
          href={this.props.serverURI}
          target="_blank"
          rel="noreferrer"
          style={{ color: 'var(--pf-global--info-color--100)' }}
        >
          {this.props.serverURI}
        </a>
        .
      </>
    );

    return (
      <CheTooltip content={content}>
        <span style={{ marginLeft: '5px' }}>
          <WarningTriangleIcon color="var(--pf-global--warning-color--100)" />
        </span>
      </CheTooltip>
    );
  }
}
