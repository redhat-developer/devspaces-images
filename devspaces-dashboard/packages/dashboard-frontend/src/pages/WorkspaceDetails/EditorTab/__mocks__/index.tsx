/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import common from '@eclipse-che/common';
import React from 'react';
import { Props, State } from '..';
import devfileApi from '../../../../services/devfileApi';
import { Workspace } from '../../../../services/workspace-adapter';

export default class EditorTab extends React.PureComponent<Props, State> {
  cancelChanges: () => void;

  constructor(props: Props) {
    super(props);

    this.state = {
      devfile: {} as devfileApi.Devfile,
      hasChanges: false,
      isDevfileValid: true,
      hasRequestErrors: false,
      currentRequestError: '',
      isExpanded: false,
      showDevfileV2ConfirmationModal: false,
    };

    this.cancelChanges = () => {
      // no-op
    };
  }

  async handleOnSave(workspace: Workspace) {
    try {
      await this.props.onSave(workspace);
    } catch (e) {
      this.setState({
        currentRequestError: common.helpers.errors.getMessage(e),
      });
    }
  }

  render() {
    return (
      <div>
        Fake Editor Tab
        <button onClick={() => this.handleOnSave(this.props.workspace)}>Save</button>
        <button onClick={() => this.props.onDevWorkspaceWarning()}>Warning</button>
        <span data-testid="current-request-error">{this.state.currentRequestError}</span>
      </div>
    );
  }
}
