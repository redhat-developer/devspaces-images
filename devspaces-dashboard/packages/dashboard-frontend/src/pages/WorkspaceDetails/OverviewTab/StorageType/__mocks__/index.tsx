/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { Props, State } from '..';

export default class StorageTypeFormGroup extends React.PureComponent<Props, State> {
  render() {
    return <div>Fake Storage Type Form</div>;
  }
}
