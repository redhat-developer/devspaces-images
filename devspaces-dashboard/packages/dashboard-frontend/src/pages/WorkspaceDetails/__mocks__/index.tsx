/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { Props, State } from '..';

export const mockShowAlert = jest.fn();
export class WorkspaceDetails extends React.PureComponent<Props, State> {
  render() {
    if (!this.props.workspace) {
      return <div>Not found</div>;
    }
    const workspace = this.props.workspace;
    return (
      <div className="WorkspaceDetails">
        Workspace Details Page
        <div data-testid="props-is-loading">{this.props.isLoading}</div>
        <div data-testid="props-old-workspace-path">
          {this.props.oldWorkspaceLocation?.pathname}
        </div>
        <div data-testid="props-show-convert-button">
          {this.props.showConvertButton === true ? 'true' : 'false'}
        </div>
        <div data-testid="props-workspace-id">{this.props.workspace.id}</div>
        <div data-testid="props-workspace-name">{this.props.workspace.name}</div>
        <div data-testid="props-workspaces-link">{this.props.workspacesLink}</div>
        <button onClick={() => this.props.onSave(workspace)}>Save</button>
        <button onClick={() => this.props.onConvert(workspace)}>Convert</button>
      </div>
    );
  }
}
