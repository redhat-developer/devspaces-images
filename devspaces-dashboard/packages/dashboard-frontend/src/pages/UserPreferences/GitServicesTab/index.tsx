/*
 * Copyright (c) 2018-2023 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { api } from '@eclipse-che/common';
import { PageSection } from '@patternfly/react-core';
import { IActionsResolver, Table, TableBody, TableHeader } from '@patternfly/react-table';
import React from 'react';
import { connect, ConnectedProps } from 'react-redux';

import ProgressIndicator from '@/components/Progress';
import { GIT_OAUTH_PROVIDERS } from '@/pages/UserPreferences/const';
import EmptyState from '@/pages/UserPreferences/GitServicesTab/EmptyState';
import GitServicesToolbar, {
  GitServicesToolbar as Toolbar,
} from '@/pages/UserPreferences/GitServicesTab/GitServicesToolbar';
import ProviderIcon from '@/pages/UserPreferences/GitServicesTab/ProviderIcon';
import ProviderWarning from '@/pages/UserPreferences/GitServicesTab/ProviderWarning';
import { AppState } from '@/store';
import * as GitOauthConfig from '@/store/GitOauthConfig';
import {
  selectGitOauth,
  selectIsLoading,
  selectProvidersWithToken,
  selectSkipOauthProviders,
} from '@/store/GitOauthConfig/selectors';

export const enabledProviders: api.GitOauthProvider[] = ['github', 'github_2'];

type Props = MappedProps;

type State = {
  selectedItems: api.GitOauthProvider[];
};

export class GitServices extends React.PureComponent<Props, State> {
  private readonly gitServicesToolbarRef: React.RefObject<Toolbar>;
  private readonly callbacks: {
    onChangeSelection?: (selectedItems: api.GitOauthProvider[]) => void;
  };

  constructor(props: Props) {
    super(props);

    this.gitServicesToolbarRef = React.createRef<Toolbar>();

    this.state = {
      selectedItems: [],
    };
  }

  private onChangeSelection(isSelected: boolean, rowIndex: number) {
    const { gitOauth } = this.props;
    if (rowIndex === -1) {
      const selectedItems = isSelected && gitOauth.length > 0 ? gitOauth.map(val => val.name) : [];
      this.setState({ selectedItems });
    } else {
      const selectedItem = gitOauth[rowIndex]?.name;
      this.setState((prevState: State) => {
        return {
          selectedItems: isSelected
            ? [...prevState.selectedItems, selectedItem]
            : prevState.selectedItems.filter(item => item !== selectedItem),
        };
      });
    }
  }

  public async componentDidMount(): Promise<void> {
    const { isLoading, requestGitOauthConfig } = this.props;
    if (!isLoading) {
      await requestGitOauthConfig();
    }
  }

  private buildGitOauthRow(gitOauth: api.GitOauthProvider, serverUrl: string): React.ReactNode[] {
    const hasWarningMessage = this.isDisabled(gitOauth) && this.hasOauthToken(gitOauth);

    const name = (
      <span key={gitOauth}>
        {GIT_OAUTH_PROVIDERS[gitOauth]}
        {hasWarningMessage && <ProviderWarning serverURI={serverUrl} />}
      </span>
    );

    const server = (
      <span key={serverUrl}>
        <a href={serverUrl} target="_blank" rel="noreferrer">
          {serverUrl}
        </a>
      </span>
    );

    const authorization = (
      <span key="Token">
        <ProviderIcon gitProvider={gitOauth} />
      </span>
    );

    return [name, server, authorization];
  }

  private isDisabled(providerName: api.GitOauthProvider): boolean {
    return (
      enabledProviders.includes(providerName) === false ||
      this.hasOauthToken(providerName) === false
    );
  }

  private isSkipOauth(providerName: api.GitOauthProvider): boolean {
    return this.props.skipOauthProviders.includes(providerName);
  }

  private hasOauthToken(providerName: api.GitOauthProvider): boolean {
    return this.props.providersWithToken.includes(providerName);
  }

  render(): React.ReactNode {
    const { isLoading, gitOauth } = this.props;
    const { selectedItems } = this.state;
    const columns = ['Name', 'Server', 'Authorization'];
    const rows =
      gitOauth.length > 0
        ? gitOauth.map(provider => {
            const canRevoke = !this.isDisabled(provider.name);
            const canClear = this.isSkipOauth(provider.name);
            return {
              cells: this.buildGitOauthRow(provider.name, provider.endpointUrl),
              selected: selectedItems.includes(provider.name),
              disableSelection: !canRevoke,
              disableActions: !canRevoke && !canClear,
              isValid: !this.isSkipOauth(provider.name),
            };
          })
        : [];

    const actionResolver: IActionsResolver = rowData => {
      if (!rowData.isValid) {
        return [
          {
            title: 'Clear',
            onClick: (event, rowIndex) => {
              event.stopPropagation();
              this.props.deleteSkipOauth(gitOauth[rowIndex].name);
            },
          },
        ];
      }
      return [
        {
          title: 'Revoke',
          onClick: (event, rowIndex) => {
            event.stopPropagation();
            this.gitServicesToolbarRef.current?.showOnRevokeGitOauthModal(rowIndex);
          },
        },
      ];
    };

    return (
      <React.Fragment>
        <ProgressIndicator isLoading={isLoading} />
        <PageSection>
          {rows.length === 0 ? (
            <EmptyState text="No Git Services" />
          ) : (
            <React.Fragment>
              <GitServicesToolbar
                ref={this.gitServicesToolbarRef}
                callbacks={this.callbacks}
                selectedItems={selectedItems}
              />
              <Table
                cells={columns}
                actionResolver={actionResolver}
                areActionsDisabled={rowData => !!rowData.disableActions}
                rows={rows}
                onSelect={(event, isSelected, rowIndex) => {
                  event.stopPropagation();
                  this.onChangeSelection(isSelected, rowIndex);
                }}
                canSelectAll={false}
                aria-label="Git services"
                variant="compact"
              >
                <TableHeader />
                <TableBody />
              </Table>
            </React.Fragment>
          )}
        </PageSection>
      </React.Fragment>
    );
  }
}

const mapStateToProps = (state: AppState) => ({
  gitOauth: selectGitOauth(state),
  providersWithToken: selectProvidersWithToken(state),
  skipOauthProviders: selectSkipOauthProviders(state),
  isLoading: selectIsLoading(state),
});

const connector = connect(mapStateToProps, GitOauthConfig.actionCreators);

type MappedProps = ConnectedProps<typeof connector>;
export default connector(GitServices);
