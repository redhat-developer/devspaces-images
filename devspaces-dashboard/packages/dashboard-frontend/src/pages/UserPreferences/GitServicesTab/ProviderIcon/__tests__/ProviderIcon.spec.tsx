/*
 * Copyright (c) 2018-2023 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { api } from '@eclipse-che/common';
import React from 'react';
import { Provider } from 'react-redux';
import { AnyAction } from 'redux';
import { MockStoreEnhanced } from 'redux-mock-store';
import { ThunkDispatch } from 'redux-thunk';

import { ProviderIcon } from '@/pages/UserPreferences/GitServicesTab/ProviderIcon';
import getComponentRenderer from '@/services/__mocks__/getComponentRenderer';
import { AppState } from '@/store';
import { FakeStoreBuilder } from '@/store/__mocks__/storeBuilder';
import {
  selectProvidersWithToken,
  selectSkipOauthProviders,
} from '@/store/GitOauthConfig/selectors';

const { createSnapshot } = getComponentRenderer(getComponent);

function getComponent(
  store: MockStoreEnhanced<AppState, ThunkDispatch<AppState, undefined, AnyAction>>,
  gitOauth: api.GitOauthProvider,
): React.ReactElement {
  const state = store.getState();
  return (
    <Provider store={store}>
      <ProviderIcon
        gitProvider={gitOauth}
        providersWithToken={selectProvidersWithToken(state)}
        skipOauthProviders={selectSkipOauthProviders(state)}
        requestSkipAuthorizationProviders={jest.fn()}
        requestGitOauthConfig={jest.fn()}
        revokeOauth={jest.fn()}
        deleteSkipOauth={jest.fn()}
      />
    </Provider>
  );
}

describe('ProviderIcon component', () => {
  test('snapshot for the successfully authorized provider', () => {
    const gitOauth: api.GitOauthProvider = 'github';
    const store = new FakeStoreBuilder().withGitOauthConfig([], ['github'], []).build();

    const snapshot = createSnapshot(store, gitOauth);

    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  test('snapshot for rejected provider', () => {
    const gitOauth: api.GitOauthProvider = 'github';
    const store = new FakeStoreBuilder().withGitOauthConfig([], [], ['github']).build();

    const snapshot = createSnapshot(store, gitOauth);

    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  test('snapshot for the provider not authorized yet', () => {
    const gitOauth: api.GitOauthProvider = 'github';
    const store = new FakeStoreBuilder().withGitOauthConfig([], [], []).build();

    const snapshot = createSnapshot(store, gitOauth);

    expect(snapshot.toJSON()).toMatchSnapshot();
  });
});
