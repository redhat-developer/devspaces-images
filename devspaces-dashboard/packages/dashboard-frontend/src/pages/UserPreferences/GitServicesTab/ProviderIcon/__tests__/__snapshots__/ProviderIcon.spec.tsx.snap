// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProviderIcon component snapshot for rejected provider 1`] = `
<div>
  <svg
    aria-hidden={true}
    aria-labelledby={null}
    fill="var(--pf-global--warning-color--100)"
    height="1em"
    role="img"
    style={
      {
        "verticalAlign": "-0.125em",
      }
    }
    viewBox="0 0 576 512"
    width="1em"
  >
    <path
      d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
    />
  </svg>
  <span>
    Authorization has been rejected by user.
  </span>
</div>
`;

exports[`ProviderIcon component snapshot for the provider not authorized yet 1`] = `
<div>
  <svg
    aria-hidden={true}
    aria-labelledby={null}
    fill="var(--pf-global--disabled-color--100)"
    height="1em"
    role="img"
    style={
      {
        "verticalAlign": "-0.125em",
      }
    }
    viewBox="0 0 1024 1024"
    width="1em"
  >
    <path
      d="M512,896 C300.2,896 128,723.9 128,512 C128,300.3 300.2,128 512,128 C723.7,128 896,300.2 896,512 C896,723.8 723.7,896 512,896 L512,896 Z M512.1,0 C229.7,0 0,229.8 0,512 C0,794.3 229.8,1024 512.1,1024 C794.4,1024 1024,794.3 1024,512 C1024,229.7 794.4,0 512.1,0 L512.1,0 Z"
    />
  </svg>
  <span>
    User has not been authorized yet.
  </span>
</div>
`;

exports[`ProviderIcon component snapshot for the successfully authorized provider 1`] = `
<div>
  <svg
    aria-hidden={true}
    aria-labelledby={null}
    fill="var(--pf-global--success-color--100)"
    height="1em"
    role="img"
    style={
      {
        "verticalAlign": "-0.125em",
      }
    }
    viewBox="0 0 512 512"
    width="1em"
  >
    <path
      d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
    />
  </svg>
  <span>
    User has been authorized successfully.
  </span>
</div>
`;
